### DISCUSSION TITLE
How to change a background image

### DISCUSSION THREAD

**Discussion by dawigit** [dawigit/picoclock#9] [Q&A] [answered]
## Now the step-by-step guide:
- install imagemagick (sudo apt install imagemagick)
- download https://en.wikipedia.org/wiki/Homer_Simpson#/media/File:Homer_Simpson_2006.png
- the png image is 190x390 pixel in size -> our goal is 256x256 in size
`cd ~/picoclock/img`
`cp ~/Downloads/Homer_Simpson_2006.png hosi2006.png`

### now we convert the image from 'alpha-mode' (checker-board background when viewed with eog) to non-alpha:

`convert hosi2006.png -transparent magenta -background black -alpha background -alpha off hosi2006a.png`

### next step is changing the size so it is quadratic

`convert hosi2006a.png -extent 390x390 -roll +88 hosi.png`

### now we shrink J to a size of 256x256

`convert hosi.png -resize 256x256 hosi256.png'

### and finally convert png to raw (img2data script in picoclock/tool/)

`img2data hosi256.png 256*256`

This gives us 'hosi256.h' , which we now have to include in the source.
We will now substitute some image in the source with our new hosi256 background.

So in 'main.c' search for 'bg4_256.h'.

The first location should be at line 56:
`#include "img/bg4_256.h"`

change it to:
`#include "img/hosi256.h"`

we continue searching with 'bg4_256' (WITHOUT .h)
and should now be at line 653:

`bg1_256,bg2_256,bg3_256,bg4_256`

change it to
`bg1_256,bg2_256,bg3_256,hosi256`

save!

### done

now build (not in DEVMODE, there are only 3 backgrounds in DEVMODE)
and flash the image. 
in pigsh: 
`bg 8`

**Comment by gamemaster7720** [comment]
do i need a older version for this because i have no 

bg1_256,bg2_256,bg3_256,bg4_256

and when i try to convert it to raw i get 

matti@matti-HP-Pavilion-Laptop-15-cs0xxx:~/Downloads/picoclock-main/tool$ img2data hosi256.png 256*256
img2data: command not found

**Reply by dawigit** [reply, marked as answer]
1. stop bugging!
2. learn linux
3. learn c
